app/navigation/nonsipp/TotalValueQuotedSharesNavigator.scala

```diff
@@ -16,19 +16,31 @@
 
 package navigation.nonsipp
 
-import models.UserAnswers
+import models.{CheckMode, NormalMode, UserAnswers}
 import navigation.JourneyNavigator
 import pages.Page
-import pages.nonsipp.totalvaluequotedshares.TotalValueQuotedSharesPage
+import pages.nonsipp.totalvaluequotedshares._
 import play.api.mvc.Call
 
 object TotalValueQuotedSharesNavigator extends JourneyNavigator {
 
-  override def normalRoutes: UserAnswers => PartialFunction[Page, Call] = userAnswers => {
+  override def normalRoutes: UserAnswers => PartialFunction[Page, Call] = _ => {
 
     case TotalValueQuotedSharesPage(srn) =>
+      controllers.nonsipp.totalvaluequotedshares.routes.TotalValueQuotedSharesCYAController.onPageLoad(srn, NormalMode)
+
+    case TotalValueQuotedSharesCYAPage(srn) =>
       controllers.nonsipp.routes.TaskListController.onPageLoad(srn)
+
   }
 
-  override def checkRoutes: UserAnswers => UserAnswers => PartialFunction[Page, Call] = _ => _ => PartialFunction.empty
+  override def checkRoutes: UserAnswers => UserAnswers => PartialFunction[Page, Call] =
```

Comment:
I don't think there is a need to have both normal mode and check mode for the CYA page here, as we only have one input field.
The input page can navigate back to the CYA page in NormalMode.
Happy to discuss if needed.